Hazel/src/Hazel/Core.h

```diff
@@ -34,10 +34,14 @@
 
 namespace Hazel {
 
-	template<typename T>
-	using Scope = std::unique_ptr<T>;
-
-	template<typename T>
-	using Ref = std::shared_ptr<T>;
+    template<typename T>
+    const auto MakeScope = std::make_unique<T>;
```

Comment:
This implementation does not currently compile for me. From my research it turns out that aliasing template functions is not the easiest thing to do. I am by no means experienced in using templates, but this seems to work:
```c++
template<typename T, typename... Args>
constexpr inline Scope<T> MakeScope(Args&&... args)
{
	return std::make_unique<T>(std::forward<Args>(args)...);
}
```
Basically instead of a straight alias, we are forwarding the function call. This might have a performance cost, but probably not a significant one. Of course this also needs to be placed after the `Scope` aliasing and then you can do the same for `Ref`, but with `MakeRef` and `make_shared`.

This compiles, but I am not experienced enough to know whether this is safe for every possible type or how it exactly works. I would like someone more experienced confirm that this is a valid design before merging.

Reply:
```cpp
	template<typename T>
	using MakeScope = std::make_unique<T>;
	template<typename T>
	using Scope = std::unique_ptr<T>;

	template<typename T>
	using MakeRef = std::make_shared<T>;
	template<typename T>
	using Ref = std::shared_ptr<T>;
```

Also: convert spaces to tabs to be inline with our coding style.